rclcpp/src/rclcpp/duration.cpp

```diff
@@ -182,6 +182,12 @@ Duration::operator-(const rclcpp::Duration & rhs) const
     rcl_duration_.nanoseconds - rhs.rcl_duration_.nanoseconds);
 }
 
+Duration
+Duration::operator*(double scale) const
+{
+  return Duration(rcl_duration_.nanoseconds * scale);
```

Comment:
Please add underflow and overflow checking here. We are checking + and - operations we should check * too.